src/gui/CMakeLists.txt

```diff
@@ -258,6 +258,10 @@ if (NOT DEFINED APPLICATION_ICON_NAME)
     set(APPLICATION_ICON_NAME ${APPLICATION_SHORTNAME})
 endif()
 
+if(NOT DEFINED APPLICATION_FOLDER_ICON_INDEX)
```

Comment:
this is somehow not working as the generated `src/gui/configgui.h` file does not have the variable defined
can you check ?

Reply:
@mgallien Not sure I follow you? How isn't this working? The generated` configgui.h `does contain this variable and everything works as expected

Reply:
that was not my case when I tried it last week
going to check again

Reply:
this all looks like something is not working as it should because you always define `APPLICATION_FOLDER_ICON_INDEX` and then the c++ code is checking if the variable is defined and in case it is undefined fallback to `0` as if that `if` was not working
can you clarify which default value is needed ?
the one in `cmake` file or the one in c++ file ?

Reply:
see https://github.com/nextcloud/desktop/pull/4014/files#diff-ee78705f6af4f851ffccb620272df3013950f876575eb559ef99ba96016e60abR55

Reply:
@mgallien The one in `cmake` is needed as default, because, if we don't have more icons in the .exe, we will fall back to the first one with index `0`, hence, `0` should be the default value. Does it make sense?

Reply:
too bad that 0 is considered as not defining the variable when `cmake` processes `#cmakedefine`